internal/subject/database/repository.go

```diff
@@ -1,84 +1,116 @@
 package database
 
 import (
+	"fmt"
+
 	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
 
 	"github.com/sumelms/microservice-course/internal/subject/domain"
 	"github.com/sumelms/microservice-course/pkg/errors"
 )
 
-type Repository struct {
-	*sqlx.DB
+const (
+	listSubject    = "list subject"
+	getSubject     = "get subject by uuid"
+	delecteSubject = "delete subject by uuid"
+)
+
+func NewRepository(db *sqlx.DB) (repository, error) {
+	sqlStatements := make(map[string]*sqlx.Stmt)
+
+	for queryName, query := range queries() {
+		stmt, err := db.Preparex(string(query))
+		if err != nil {
+			return repository{}, errors.WrapErrorf(err, errors.ErrCodeUnknown, "error preparing statement %s", queryName)
+		}
+		sqlStatements[queryName] = stmt
+	}
+	fmt.Println(sqlStatements)
```

Comment:
I think you leave it for debuging/testing proposal, but... if we want to keep this as a message, make it a log as info level.

Reply:
Yes, it was for debugging purposes and I forgot to remove it.